ci/prow/make_config.go

```diff
@@ -1302,6 +1306,23 @@ func setOutput(fileName string) {
 	output = configFile
 }
 
+// Recursively finding all yaml files under "asis" dir, add them asis in prow jobs
+func generateAsisJobs() {
```

Comment:
You should be able to generate this job as a custom job plus arguments override.

Like https://github.com/knative/test-infra/blob/master/ci/prow/config_knative.yaml#L121

Reply:
This is just an attempt of trying out ways of supporting very custom jobs. I will update this PR to use arguments override

Reply:
I understand. But I would very much prefer having a framework around the jobs so all pieces are in sync, instead of going back to having several fully expanded job definitions (I know this is just one job, but that'll make it easy to start having copy&paste jobs added to Prow).

Reply:
Totally agreed, keeping the base in sync for future fully expanded jobs is the biggest con for this approach. I will explore other ways in future works

Reply:
done